prototyping/xarray/xarray for dose.ipynb

```diff
@@ -0,0 +1,263 @@
+{
```

Comment:
<p>While it might increase scope dramatically, it's useful to have an object representing the orientation be displayed as well (e.g. a "humaniform phantom outline" or a Cube with the S/I/A/P/L/R shown on the appropriate part of the surface).</p><p>And it's important to identify what variable comes in to play that causes one reference frame to no longer be the same as a parent reference frame (e.g. IEC 61217 (a.k.a. Table Top) ceases to be the same as IEC FIXED when the Couch Angle (Patient Support Angle) is no longer 0 degrees.  Robot tables in the room prefer to "think" in IEC FIXED.  But RT Plan is in terms of IEC 61217.  That these are not the same doesn't become apparent until the couch is no longer at 0 degrees.  </p><p>If your purpose is to visualise the dose to a water phantom *as if one were looking at the water phantom*, IEC FIXED is probably fine.  If your purpose is to visualise the dose to a water phantom *as if it was dose in the patient while looking at the patient on the table*, you may want to map to IEC 61217.  But if you want to view it as if it is on the images of the patient, then you may want to use DICOM (and not IEC PATIENT).</p><p>The moment you start expressing concern about the patient's orientation, you aren't really dealing with IEC FIXED any more.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pymedphys/pymedphys/pull/1430/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='19'/>

Reply:
> While it might increase scope dramatically, it's useful to have an object representing the orientation be displayed as well (e.g. a "humaniform phantom outline" or a Cube with the S/I/A/P/L/R shown on the appropriate part of the surface).

Agree, this is just a start to demo XArray. I actually have taken a nice set of 8 CT scans of a coffee cup at each of the orthogonal orientations. In each scan, the coffee cup is oriented according to the scan orientation, where the cup opening represents the face direction and the handle represents the head direction. I have corresponding RTDOSES for these scans, and I'd hoped to use them for demoing exactly this. I'm just not sure of the latest protocol for test data in PyMedPhys. Where would I upload said doses? Are they going to be too large? Actually, their empty equivalents (PixelData wiped) are already in the testing data from a couple of years ago.

> And it's important to identify what variable comes in to play that causes one reference frame to no longer be the same as a parent reference frame (e.g. IEC 61217 (a.k.a. Table Top) ceases to be the same as IEC FIXED when the Couch Angle (Patient Support Angle) is no longer 0 degrees. Robot tables in the room prefer to "think" in IEC FIXED. But RT Plan is in terms of IEC 61217. That these are not the same doesn't become apparent until the couch is no longer at 0 degrees.

Excellent point, I hadn't considered table top rotation. I'd be in favour of replacing IEC FIXED with IEC 61217.

I expect that these objects are more likely to be used to view patient doses (as opposed to water phantom doses).

Reply:
Let me elaborate a little on how I'd like to be able to view the dose here. It helps to conceptually separate the pixel grid from its corresponding coordinates. Let's say that the grid is indexed as `grid[k, i, j]`

Imagine we're viewing the dose to a patient lying on the table top in the transverse. The patient is HFS, the table top is 0, gantry at 0 (up). I'd like for the user to be looking at a grid that is depicted as though the user were looking at the patient from the far end of the couch (inf-end in this case). The key is: I'd like for this to *always* be the case, regardless of patient orientation. I.e., `grid[0, 0, 0]` is at the upper left corner of the image and farthest from the gantry, and in IEC Table Top, `k` increases with `y` , `i` _decreases_ with `z` and `j` increases with `x`.

I'd like to provide support for the user to view the grid with axes labelled according to the IEC Patient coordinate system as well as IEC Table Top, but in this case, they match anyway

Hopefully that makes sense.

Now, if we are to consider a new orientation, say FFP, my intention is for the grid _not to rotate according to the patient orientation_. I.e., the patient appears face down in the grid and `k` would now decrease with `z` in IEC Table Top. In IEC Table Top, this would mean that the axis labels would match (assuming same extent). But in IEC patient, some of the axis labels would *change in sign*. So, it's still as if we're looking from the far end of the couch, but we have the option to *label* the axes according to different coord systems; namely e.g. DICOM, IEC PATIENT, IEC TABLE TOP (IEC FIXED support would be complicated and I think fine to ignore).

Does that all make sense?

Incidentally, RayStation depicts the patient as I've described above but with axis labels in IEC Patient. But I have seen some viewers *rotate the grid with orientation!* I feel like this is so counterintuitive for most, it's almost a bug.

Reply:
The preferred view you describe is IEC61217.  You are viewing it from the foot of the Table Top.

It makes sense to me.
That is the perspective of the clinician observing the patient, more or less.
It's the view a Radiologist takes of a CT.